Core.Collectors/IO/AzureHelpers.cs

```diff
@@ -269,5 +281,51 @@ public static string GenerateNotificationMessage(CloudBlob blob)
             string path = blobPath.Split($"{container}/").Last();
             return Tuple.Create(container, path);
         }
+
+        public static CloudStorageAccount GetStorageAccount(string storageConnectionEnvironmentVariable)
+        {
+            string stagingBlobConnectionString = Environment.GetEnvironmentVariable(storageConnectionEnvironmentVariable);
+            return CloudStorageAccount.Parse(stagingBlobConnectionString);
+        }
+
+        public async Task<CloudStorageAccount> GetStorageAccountUsingMsiAsync(string storageAccountNameEnvironmentVariable, bool isQueue = false)
+        {
+            string storageAccountName = Environment.GetEnvironmentVariable(storageAccountNameEnvironmentVariable);
+            string resource = isQueue ? GetQueueResource(storageAccountName) : GetBlobResource(storageAccountName);
+            StorageCredentials creds = await GetStorageCredentails(resource).ConfigureAwait(false);
+            CloudStorageAccount storageAccount = new CloudStorageAccount(creds, storageAccountName, EndPointSuffix, true);
+            return storageAccount;
+        }
+
+        private async Task<StorageCredentials> GetStorageCredentails(string resource)
+        {
+            AzureServiceTokenProvider azureServiceTokenProvider = new AzureServiceTokenProvider();
+            string token = await azureServiceTokenProvider.GetAccessTokenAsync(resource).ConfigureAwait(false);
+            var jwt = new JwtSecurityToken(token);
+            this.tokenExpiration = jwt.ValidTo;
+            Dictionary<string, string> properties = new Dictionary<string, string>()
+            {
+                { "Resource", resource },
+                { "TokenValidTo", tokenExpiration.ToString()}
+            };
+            this.telemetryClient.TrackEvent("MsiTokenGeneration", properties);
```

Comment:
This is not how you want to do this, believe me :)

I will propose something, let me know if that does not work.

Instead of changing the static-ness of this class (leave it as static), pass ITelemetryClient as a required (or optional?) parameter to every single method that needs to access it. This way, you don't need to make this class an instance class and deal with state.